wai-extra/Network/Wai/Middleware/CombineHeaders.hs

```diff
@@ -0,0 +1,245 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE RecordWildCards #-}
+{- |
+Sometimes incoming requests don't stick to the
+"no duplicate headers" invariant, for a number
+of possible reasons (e.g. proxy servers blindly
+adding headers), or your application (or other
+middleware) blindly adds headers.
+
+In those cases, you can use this 'Middleware'
+to make sure that headers that /can/ be combined
+/are/ combined. (e.g. applications might only
+check the first \"Accept\" header and fail, while
+there might be another one that would match)
+    -}
+module Network.Wai.Middleware.CombineHeaders
+    ( combineHeaders
+    , defaultCombineSettings
+    , setHeader
+    , removeHeader
+    , CombineSettings (..)
```

Comment:
By exposing the data constructors for these two types, it means that any additions in the future will be breaking changes. I'd rather follow the smart constructor/settings type approach.

Reply:
Will change this next time I can focus on this PR 👍

Reply:
Made new commit that added the smart constructor approach.